### PR TITLE
Reduce update rate of viewer count

### PR DESCRIPTION
**Pull request by sphinxrave**

### CODE REVIEW

File: apps/crawler-youtube/tasks/video-info-api.js [hunk 1]
```diff
@@ -122,8 +122,9 @@ module.exports = async () => {
           saveInfo.live_schedule = ytInfo.liveStreamingDetails.scheduledStartTime || null;
           saveInfo.live_start = ytInfo.liveStreamingDetails.actualStartTime || null;
           saveInfo.live_end = ytInfo.liveStreamingDetails.actualEndTime || null;
+          const oldViewerCount = targetVideo.live_viewers;
           const currentViewers = +ytInfo.liveStreamingDetails.concurrentViewers || null;
-          saveInfo.live_viewers = currentViewers && +(currentViewers.toPrecision(currentViewers > 1000 ? 2 : 1));
+          saveInfo.live_viewers = currentViewers && updateIfSignificant(oldViewerCount, +(currentViewers.toPrecision(currentViewers > 1000 ? 2 : 1)), 300);
```
Comment (IAmVisco) [inline apps/crawler-youtube/tasks/video-info-api.js:127]: Any reason to use `toPrecision`? It's for the double values while I doubt we can have 0.5 viewers.

File: modules/utils.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+function hashCode(s) {
```
Comment (IAmVisco) [inline modules/utils.js:1]: Is it even used?

File: modules/utils.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  let h;
+  // eslint-disable-next-line no-bitwise
+  for (let i = 0; i < s.length; i += 1) h = Math.imul(31, h) + s.charCodeAt(i) | 0;
+  return h;
+}
+
+function updateIfSignificant(oldNumber, newNumber, minDelta) {
+  if (oldNumber == null) return newNumber;
```
Comment (IAmVisco) [inline modules/utils.js:9]: Don't you need eslint-ignore over double eq?

**Review by IAmVisco — CHANGES_REQUESTED** [review]

File: apps/crawler-youtube/tasks/video-info-api.js (hunk 1 above)
Comment (IAmVisco) [inline apps/crawler-youtube/tasks/video-info-api.js:127]: 300 is being duplicated somewhere and looks like something that should be configured without redeploy. Mind moving it to ENV var? Make sure to read it only once on app start though to reduce env syscalls.

**Review by IAmVisco — CHANGES_REQUESTED** [review]

File: apps/crawler-youtube/tasks/video-info-api.js (hunk 1 above)
Comment (sphinxrave) [reply apps/crawler-youtube/tasks/video-info-api.js:127]: ok

**Review by IAmVisco — APPROVED** [review]

**Pull request merged**